--- .codegen/src/main/java/com/taurushq/sdk/protect/openapi/ApiClient.java	2024-10-10 17:09:47
+++ .codegen/src/main/java/com/taurushq/sdk/protect/openapi/ApiClient2.java	2024-10-10 17:29:55
@@ -51,10 +51,7 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import com.taurushq.sdk.protect.openapi.auth.Authentication;
-import com.taurushq.sdk.protect.openapi.auth.HttpBasicAuth;
-import com.taurushq.sdk.protect.openapi.auth.HttpBearerAuth;
-import com.taurushq.sdk.protect.openapi.auth.ApiKeyAuth;
+import com.taurushq.sdk.protect.openapi.auth.*;
 
 /**
  * <p>ApiClient class.</p>
@@ -100,7 +97,7 @@
         initHttpClient();
 
         // Setup authentications (key: authentication name, value: authentication).
-        authentications.put("ApiKeyTPV1", new ApiKeyAuth("header", "Authorisation"));
+        authentications.put("ApiKeyTPV1", new ApiKeyTPV1Auth());
         // Prevent the authentications from being modified.
         authentications = Collections.unmodifiableMap(authentications);
     }
@@ -116,7 +113,7 @@
         httpClient = client;
 
         // Setup authentications (key: authentication name, value: authentication).
-        authentications.put("ApiKeyTPV1", new ApiKeyAuth("header", "Authorisation"));
+        authentications.put("ApiKeyTPV1", new ApiKeyTPV1Auth());
         // Prevent the authentications from being modified.
         authentications = Collections.unmodifiableMap(authentications);
     }
@@ -147,6 +144,36 @@
     }
 
     /**
+     * Helper method to set API key value for the first API key authentication.
+     *
+     * @param apiKey API key
+     */
+    public void setApiKeyTPV1(String apiKey) {
+        for (Authentication auth : authentications.values()) {
+            if (auth instanceof ApiKeyTPV1Auth) {
+                ((ApiKeyTPV1Auth) auth).setApiKey(apiKey);
+                return;
+            }
+        }
+        throw new RuntimeException("No API key authentication configured!");
+    }
+
+    /**
+     * Helper method to set API secret value for the first API key authentication.
+     *
+     * @param apiSecret API key
+     */
+    public void setApiSecretTPV1(String apiSecret) throws ApiKeyTPV1Exception {
+        for (Authentication auth : authentications.values()) {
+            if (auth instanceof ApiKeyTPV1Auth) {
+                ((ApiKeyTPV1Auth) auth).setApiSecret(apiSecret);
+                return;
+            }
+        }
+        throw new RuntimeException("No API secret authentication configured!");
+    }
+
+    /**
      * Get base path
      *
      * @return Base path
